### DISCUSSION TITLE
Foundry Fundamentals, Section 4 Raffle, Lesson 32 fund subscription, invalid type for function parameter

### DISCUSSION THREAD

**Discussion by BowtiedHarpyEagle** [Cyfrin/foundry-full-course-cu#2648] [Q&A]
I am going through the latest version of this foundry fundamentals course on Updraft. I am getting this error interactions.s.sol at about 10:00 timestamp on lesson 32 

"invalid type for argument in function call" apparently, the subscriptionId is uint256 while the VRFCoordinatorV2Mock requires type uint64 as an input. 

You can see all my code in the repository I use to keep track of my course progress here: 

https://github.com/BowtiedHarpyEagle/cu-ff-s4-lottery

My larger issue is that I find the new version of this course quite confusing. In some places, videos don't match text lessons which at times don't match the github resources on what to import etc... I'm not sure which course source I should use as the latest-best source of truth on course materials that work.

**Comment by EngrPips** [comment]
Hello @BowtiedHarpyEagle, We sorted this on Discord already. correct?

**Reply by BowtiedHarpyEagle** [reply]
Yes. It is solved after I have corrected the imported VRF coordinator mock contract to version 2.5.

**Reply by EngrPips** [reply]
You are giga chad, Keep crushing it.
